dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -353,9 +354,19 @@ else if ( maxLimit > 0 )
         // Grid
         // ---------------------------------------------------------------------
 
+        String transformedSql = null;
         try
         {
-            getAggregatedEventData( grid, params, sql );
+            transformedSql = SqlStringProcessor.toInnerJoins( sql, false );
+        }
+        catch ( Exception e )
+        {
+            log.debug( sql );
```

Comment:
If I understand it correctly, in case something fails in SQL transformation it still will use un-transformed SQL.
It's fine for resilience but if SQL transformation doesn't happen for any reason, we won't even notice it.